pymc3/distributions/multivariate.py

```diff
@@ -372,8 +372,8 @@ def __init__(self, a, transform=transforms.stick_breaking,
         kwargs.setdefault("shape", shape)
         super(Dirichlet, self).__init__(transform=transform, *args, **kwargs)
         
-        self.k = tt.as_tensor_variable(shape)
-        self.a = a = tt.as_tensor_variable(a)
+        self.k = floatX(shape)
+        self.a = a = floatX(a)
```

Comment:
total nit, and not necessary, but might be nicer to just use `self.a` below, and stop referring to `a` (same with `Sigma` and `nu`.